sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -987,6 +987,24 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       checkAnswer(agg, ds.groupBy('id % 2).agg(count('id)))
     }
   }
+
+  test("array") {
```

Comment:
can you think of a better test name which can describe what we are testing here?